scripts/xdist/find_order_dependent_test_failures.py

```diff
@@ -0,0 +1,221 @@
+    """
+    global fast_option
+    worker_test_dict = {}
+    test_base_included = {}
+    failing_worker_num = None
+    with io.open(log_file, 'r') as console_file:
+        for line in console_file:
+            regex_search = re.search(r'\[gw(\d+)] (PASSED|FAILED|SKIPPED|ERROR) (\S+)'.format(test_suite), line)
+            if regex_search:
+                worker_num_string = regex_search.group(1)
+                pass_fail_string = regex_search.group(2)
+                if worker_num_string not in worker_test_dict:
+                    worker_test_dict[worker_num_string] = []
+                test = regex_search.group(3)
+                if test_suite == "commonlib-unit":
+                    if "pavelib" not in test and not test.startswith('scripts'):
+                        test = u"common/lib/{}".format(test)
+                if fast_option and pass_fail_string == 'PASSED':
+                    # fast option will only take one test per class or module, in case
+                    # the failure is a setup/teardown failure.
+                    test_base = '::'.join(test.split('::')[:-1])
+                    if test_base not in test_base_included:
+                        worker_test_dict[worker_num_string].append(test)
+                        test_base_included[test_base] = True
+                elif (not fast_option or (fast_option and pass_fail_string == 'FAILED')):
+                    worker_test_dict[worker_num_string].append(test)
+                if pass_fail_string == 'FAILED':
+                    failing_worker_num = worker_num_string
+                    break
+    if failing_worker_num:
+        return worker_test_dict[failing_worker_num]
+
+
+def _get_pytest_command(output_file_name):
+    """
+    Return the pytest command to run.
+    """
+    return "pytest -p 'no:randomly' `cat {}`".format(output_file_name)
+
+
+def _run_tests_and_check_for_failures(output_file_name):
+    """
+    Runs tests and returns True if failures are found.
+    """
+    global verbose_option
+    pytest_command = _get_pytest_command(output_file_name)
+    test_output = os.popen(pytest_command).read()
+    if verbose_option:
+        print(test_output)
+    failures_search = re.search(r'=== (\d+) failed', test_output)
+    return bool(failures_search) and int(failures_search.group(1)) > 0
+
+
+def _create_and_check_test_files_for_failures(test_list, test_type):
+    """
+    Run the test list to see if there are any failures.
+
+    Keeps around any test files that produced a failure, and deletes
+    the passing files.
+
+    Returns the pytest command to run if failures are found.
+    """
+    print("Testing {}, includes {} test(s)...".format(test_type, len(test_list)))
+    global test_suite_option
+    output_file_name = "{}/{}_failing_test_list_{}_{}.txt".format(
+        OUTPUT_FOLDER_NAME, test_suite_option, test_type, len(test_list)
+    )
+    dirname, basename = os.path.split(output_file_name)
```

Comment:
Wouldn't `dirname` always be `OUTPUT_FOLDER_NAME`?

Reply:
I updated, even though it is the same number of lines splitting or joining this path.  Maybe it is more clear not introducing dirname. :)